cmd/xl-v1-object.go

```diff
@@ -412,13 +405,11 @@ func (xl xlObjects) GetObjectInfo(ctx context.Context, bucket, object string, op
 	}
 
 	if hasSuffix(object, slashSeparator) {
-		if !xl.isObjectDir(bucket, object) {
```

Comment:
@harshavardhana I believe this line is needed, this is to make the difference between an empty directory and a non empty directory. 

getObjectInfoDir() will just return directory info for every directory

Reply:
It won't be getObjectInfoDir() is changed to incorporate isObjectDir behavior as well.